lib/collection/bloc/collection_bloc.dart

```diff
@@ -69,64 +80,200 @@ class CollectionBloc extends Bloc<CollectionEvent, CollectionState> {
+  void _onInterviewStartTimeChanged(
+      InterviewStartTimeChanged event, Emitter<CollectionState> emit) {
+    final interviewStartTime =
+        InterviewStartTime.dirty(event.interviewStartTime);
     emit(state.copyWith(
-      householdId: householdId,
-      sensitizationStatus: Formz.validate([
-        householdId,
-        state.respondentName,
-        state.respondentTelNumber,
-        state.interviewDate
-      ]),
-    ));
-  }
-
-  void _onRespondentNameUnfocused(
-      RespondentNameUnfocused event, Emitter<CollectionState> emit) {
-    final respondentName = RespondentName.dirty(state.respondentName.value);
-    emit(state.copyWith(
-      respondentName: respondentName,
-      sensitizationStatus: Formz.validate([
-        state.householdId,
-        respondentName,
-        state.respondentTelNumber,
-        state.interviewDate
-      ]),
-    ));
-  }
-
-  void _onRespondentTelNumberUnfocused(
-      RespondentTelNumberUnfocused event, Emitter<CollectionState> emit) {
-    final respondentTelNumber =
-        RespondentTelNumber.dirty(state.respondentTelNumber.value);
-    emit(state.copyWith(
-      respondentTelNumber: respondentTelNumber,
-      sensitizationStatus: Formz.validate([
-        state.householdId,
-        state.respondentName,
-        respondentTelNumber,
-        state.interviewDate
-      ]),
-    ));
-  }
-
-  void _onInterviewDateUnfocused(
-      InterviewDateUnfocused event, Emitter<CollectionState> emit) {
-    final interviewDate = InterviewDate.dirty(state.interviewDate.value);
-    emit(state.copyWith(
-      interviewDate: interviewDate,
-      sensitizationStatus: Formz.validate([
-        state.householdId,
-        state.respondentName,
-        state.respondentTelNumber,
-        interviewDate
-      ]),
-    ));
+        interviewStartTime: interviewStartTime,
+        sensitizationStatus: Formz.validate([
+          state.householdId,
+          state.respondentName,
+          state.respondentTelNumber,
+          state.interviewDate,
+          interviewStartTime
+        ])));
+  }
+
+  void _onFoodItemAdded(FoodItemAdded event, Emitter<CollectionState> emit) {
+    final foodItem = FoodItem();
+    List<FoodItem> foodItems = List.from(state.foodItems);
+    foodItems.add(foodItem);
+
+    emit(state.copyWith(foodItems: foodItems));
+  }
+
+  void _onFoodItemDeleted(
+      FoodItemDeleted event, Emitter<CollectionState> emit) {
+    List<FoodItem> foodItems = List.from(state.foodItems);
+
+    // TODO: change into UUID-based indexing
+    foodItems.remove(event.foodItem);
+
+    emit(state.copyWith(foodItems: foodItems));
+  }
+
+  void _onFoodItemNameChanged(
+      FoodItemNameChanged event, Emitter<CollectionState> emit) {
+    List<FoodItem> foodItems = List.from(state.foodItems);
+
+    // TODO: change into UUID-based indexing
+    int changedFoodItemIndex = foodItems.indexOf(event.foodItem);
+
+    FoodItem foodItem = foodItems[changedFoodItemIndex]
```

Comment:
might be a good place to similarly implement un-confirming the fooditem.confirmed

Reply:
Great, just implemented it! Thanks! 🙌